tests/test_merge_all.py

```diff
@@ -274,3 +279,134 @@ def test_merge_all_on_non_overlapping_rasters_returns_first_raster():
     raster2 = make_test_raster(value=2, band_names=['blue'], affine=affine2, height=30, width=40)
     merged = merge_all([raster1, raster2], raster1.footprint())
     assert merged == raster1
+
+
+def test_merge_does_not_uncover_masked_pixels():
+    # See https://github.com/satellogic/telluric/issues/65
+    affine = Affine.translation(0, 2) * Affine.scale(1, -1)
+
+    rs_a = GeoRaster2(
+        image=np.ma.masked_array([
+            [
+                [100, 89],
+                [100, 89]
+            ],
+            [
+                [110, 99],
+                [110, 99]
+            ]
+        ], [
+            [
+                [False, True],
+                [False, True]
+            ],
+            [
+                [False, True],
+                [False, True]
+            ]
+        ], dtype=np.uint8),
+        affine=affine,
+        crs=WGS84_CRS,
+        band_names=['red', 'green'],
+    )
+
+    rs_b = GeoRaster2(
+        image=np.array([[
+            [0, 210],
+            [0, 210]
+        ]], dtype=np.uint8),
+        affine=affine,
+        crs=WGS84_CRS,
+        band_names=['green'],
+    )
+
+    expected_image = np.ma.masked_array([
+        [
+            [100, 89],
+            [100, 89]
+        ],
+        [
+            [110, 99],
```

Comment:
shouldn't green be:
[[110, 210], [110, 210]]

because of the mask of the first one?

Reply:
The right column will be masked anyway, because the second raster does not have green band. Therefore, the masked value (which shouldn't be visible anyway) isn't changed.

Reply:
but the second raster has `green`, it doesn't have `red` so for `red` I agree, but not for `green`

Reply:
Sorry, I meant red and not green. Still, masked pixels should never be unmasked, so the underlying `data` value is not important. The algorithm is just discarding it.

Reply:
> it should fill the masked pixels with pixels values from the next raster valid pixels

...unless there's not enough data to have a per-dataset mask that show those points. If I add the new pixels in the green band, I'm uncovering the masked red pixels, because I don't produce a different mask per band. That's actually issue #65.